Studio-Lovelies/GG-JointJustice-Unity
# Feature/script pre loading

## Summary

Implements a feature that reads an Ink JSON file and loads any assets required. This is handled by a new NarrativeScript class which replaces the previous Dialogue struct. Scripts can access the loaded and stored objects through the NarrativeScript class' ObjectStorage property.

This feature replaces the previous method of assigning actors and evidence available in a scene. This will allow for the game to be expanded upon more easily without requiring extensive use of the Unity editor.

ActorData, Evidence, and AudioClip objects are supported. ~~Support for loading BGScenes is planned, but want to wait for the asset normalisation PR #141 to go through.~~ BGScene preloading is now also supported.

The evidence menu tests scripts broke so took the opportunity to rework them to be a lot faster, cleaner, and less reliant on Ink scripts and Unity Scenes as well a fixing a bug where the cursor was positioned in the wrong place to click some buttons.

~~Marked as draft until: #142 is merged~~

Updated to include #142. Also closes #108 closes #109 and closes #110 because it broke the previous penalty implementation, so included those fixes when updating it.

## Testing instructions

Play scripts in Inky-TestScene and make sure they still work.
Run the associated tests.

## Additional information

- `[ ]` Changes UI
- `[x]` Introduces new feature
- `[x]` Removes existing feature
- `[ ]` Has associated resource:
